jupyterhub/templates/image-puller/serviceaccount.yaml

```diff
@@ -2,18 +2,16 @@
 ServiceAccount for the pre-puller hook's image-awaiter-job
 */}}
 {{- if .Values.prePuller.hook.serviceAccount.create -}}
-{{- if (include "jupyterhub.imagePuller.daemonset.hook.install" .) -}}
+{{- if or (include "jupyterhub.imagePuller.daemonset.hook.install" .) .Values.prePuller.continuous.enabled -}}
 apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: {{ include "jupyterhub.hook-image-awaiter-serviceaccount.fullname" . }}
   labels:
     {{- include "jupyterhub.labels" . | nindent 4 }}
-    hub.jupyter.org/deletable: "true"
   annotations:
     "helm.sh/hook": pre-install,pre-upgrade
-    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
-    "helm.sh/hook-weight": "0"
+    "helm.sh/hook-weight": "-30"
```

Comment:
This is to force the creation of a service account before the prepullers start.

Reply:
Other objects such as RBAC roles use weight `0`. If the ServiceAccount uses weight `0` does this break?

Reply:
Yes. If we use weight `0` this breaks, probably because the pre-puller hook starts before the service account is created.